backend/src/v4/models/project.js

```diff
@@ -469,7 +469,7 @@
 		}
 
 		if (data.permissions) {
-			data.permissions.forEach((permissionUpdate) => {
+			data.permissions.forEach(async (permissionUpdate) => {
```

Comment:
you'd need to turn the `forEach` into a `map` with a `Promise.all` if you change this function to async, otherwise we wouldn't be waiting on it